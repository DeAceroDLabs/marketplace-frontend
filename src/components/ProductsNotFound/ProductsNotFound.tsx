import React from "react";
import NotFound from "components/common/NotFound";
import styles from "./ProductsNotFound.module.scss";

const Search: React.FunctionComponent = () => {

  return (
    <NotFound title="No encontramos un producto que coincida con tu búsqueda" imgSrc="/not-found.jpg" 
    info="Navega por las categorías para encontrar un producto similar"/>
  );
};

export default Search;
