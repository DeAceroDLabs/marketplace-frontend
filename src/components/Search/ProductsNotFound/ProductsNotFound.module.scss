@use 'styles/colors';
@use 'styles/typography';

a:link {
  color: colors.$accent;
  background-color: transparent;
}

a:visited {
  color: colors.$accent;
  background-color: transparent;
}


a:active {
  color: colors.$accent;
  background-color: transparent;
}
