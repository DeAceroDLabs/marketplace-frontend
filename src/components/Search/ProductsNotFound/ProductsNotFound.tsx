import React from "react";
import NotFound from "components/common/NotFound";
import styles from "./ProductsNotFound.module.scss";

const Search: React.FunctionComponent = () => {
  return (
    <NotFound
      title="No encontramos un producto que coincida con tu búsqueda"
      imgSrc="/not-found.jpg"
    >
      <div>
        Navega por las{" "}
        <a className={styles.a} href="/">
          categorías
        </a>{" "}
        para encontrar un producto similar
      </div>
    </NotFound>
  );
};

export default Search;
