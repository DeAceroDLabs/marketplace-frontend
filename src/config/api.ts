import { API, Category, Product, ZipCodeLocation } from "./api.types";
import axios from "axios";
import { USER_API } from "./user.api.types";

const fetchData = (url: string) => {
  async function fetchUrl() {
    const response = await fetch(url);
    const json = await response.json();
    return json;
  }
  return fetchUrl();
};

const postRequest = async (url: string, body: any) => {
  const response = await axios.post(USER_API.createUser, body);
  return response;
};

const fetchCategories = (): Promise<Category[]> => {
  return fetchData(API.categories);
};

const fetchProducts = (categoryId: number): Promise<Product[]> => {
  return fetchData(API.products(categoryId));
};

const getLocation = (zipCode: string): Promise<ZipCodeLocation> => {
  return fetchData(API.location(zipCode)).then((data) => {
    if (data["location"]) {
      const neighborhoodsObject = data["location"]["neighborhood"];
      const neighborhoods = Object.keys(neighborhoodsObject);
      const neighborhoodsArray = neighborhoods.map((neighborhood) => {
        return {
          value: neighborhood,
          label: neighborhoodsObject[neighborhood],
        };
      });
      data["location"]["neighborhood"] = neighborhoodsArray;
    }
    return data;
  });
};

const searchProducts = (query: string): Promise<any> => {
  const body = {
    q: query,
  };
  return postRequest(API.search, body);
};

export {
  postRequest,
  fetchCategories,
  fetchProducts,
  searchProducts,
  getLocation,
};
