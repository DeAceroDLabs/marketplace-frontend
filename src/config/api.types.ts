export const API = {
  categories: "categories",
  products: (categoryId: number) => `categories/${categoryId}/products`,
  search: "/search",
  location: (zipCode: string) => `locations/postalCode/${zipCode}`,
};

export type Category = {
  categoryId: number;
  categoryName: string;
  imageUrl: string;
};

export type Article = {
  articleId: string;
};

export type Product = {
  productId: number;
  productName: string;
  categoryName: string;
  categoryId: number;
  articles: Article[];
  categoryImageUrl: string;
};

export type ZipCodeLocation = {
  countryName: string;
  location: Location;
};

export type Location = {
  cityId: string;
  city: string;
  stateId: string;
  state: string;
  neighborhood: Neighborhood[];
};

export type Neighborhood = {
  value: string;
  label: string;
};
