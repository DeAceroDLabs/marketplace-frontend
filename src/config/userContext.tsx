import React from "react";

const defaultUser = {
  username: "",
  setUser: (username: string) => {},
};

const UserContext = React.createContext(defaultUser);

export const UserProvider = UserContext.Provider;

export default UserContext;
