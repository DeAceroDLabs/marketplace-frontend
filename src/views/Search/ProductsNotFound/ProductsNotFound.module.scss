@use 'styles/colors';
@use 'styles/typography';

.categories{
  display: inline-block;
  -webkit-text-decoration-line: underline; /* Safari */
  text-decoration-line: underline; 
}
