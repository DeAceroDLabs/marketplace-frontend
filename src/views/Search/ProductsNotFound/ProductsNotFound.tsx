import React from "react";
import NotFound from "components/common/NotFound";
import styles from "./ProductsNotFound.module.scss";
import Button from "components/common/Button";
import { useNavigate } from "react-router-dom";

const Search: React.FunctionComponent = () => {
  const navigate = useNavigate();
  return (
    <NotFound
      title="No encontramos un producto que coincida con tu búsqueda"
      imgSrc="/product-not-found.jpg"
    >
      <div>
        Navega por las
        <div className={styles.categories}>
        <Button color="white"  action={() => navigate("/")}>
          categorías
        </Button>
        </div>
        para encontrar un producto similar
      </div>
    </NotFound>
  );
};

export default Search;
