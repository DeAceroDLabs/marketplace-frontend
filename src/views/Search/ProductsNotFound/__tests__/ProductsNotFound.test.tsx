import renderer from "react-test-renderer";
import ProductsNotFound from "../ProductsNotFound";
import { MemoryRouter } from "react-router-dom";
import { UserProvider } from "config/userContext";

const mockedUsedNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...(jest.requireActual("react-router-dom") as any),
  useNavigate: () => mockedUsedNavigate,
}));

describe("Header", () => {
  const mockUSer = {
    username: "mock_user",
    setUser: jest.fn(),
  };
  it("renders ProductsNotFound with no issue", () => {
    <UserProvider value={mockUSer}>
      <MemoryRouter>
        <ProductsNotFound />
      </MemoryRouter>
    </UserProvider>;
    const view = renderer.create(<ProductsNotFound />);
    expect(view).toMatchSnapshot();
  });
});
