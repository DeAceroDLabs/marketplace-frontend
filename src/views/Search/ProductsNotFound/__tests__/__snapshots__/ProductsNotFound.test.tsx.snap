// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders ProductsNotFound with no issue 1`] = `
<div>
  <div
    className="container"
  >
    <img
      alt="Not found img"
      className="image"
      src="/product-not-found.jpg"
    />
    <div
      className="containerInfo"
    >
      <div
        className="title"
      >
        No encontramos un producto que coincida con tu búsqueda
      </div>
      <div
        className="info"
      >
        <div>
          Navega por las
          <div
            className="categories"
          >
            <button
              className="button white"
              onClick={[Function]}
            >
              categorías
            </button>
          </div>
          para encontrar un producto similar
        </div>
      </div>
    </div>
  </div>
</div>
`;
